### PR TITLE
More granular restriction on resource deletion

### PR DESCRIPTION
**Pull request by tea-short**
* Create a policy per account instead of a singular restriction policy attached at the root account level and inherited by all accounts. 
Exceptions are set so that deleting KMS keys and Route53 PHZs is not denied to:
    * `breakglass` user in any account
    * `ci` role in any account
    * `administrator` role in any account (where it is present, i.e. `dev` & `mgmt-dev`)

* Deleting CloudWatch logs and flow logs is no longer denied to `breakglass` user in any account.
* S3 encryption enforcement dropped

### CODE REVIEW

**Comment by djsd123** [comment]
Have you deployed this yet?

File: locals.tf [hunk 1]
```diff
@@ -0,0 +1,20 @@
+data "aws_caller_identity" "root_account" {}
+
+locals {
```
Comment (djsd123) [inline locals.tf:3]: If we have to maintain a list of account IDs then it should be noted: https://github.com/IDPdigital/idp-accounts-config#usage

File: locals.tf [hunk 2]
```diff
@@ -0,0 +1,20 @@
+data "aws_caller_identity" "root_account" {}
+
+locals {
+
+  account_ids = {
```
Comment (djsd123) [inline locals.tf:5]: Also, we export member account ids among other ids which would be easier to manage in the long run. https://github.com/IDPdigital/idp-accounts-config#outputs

**Comment by djsd123** [comment]
Personally, I think policies should apply to all accounts in the org and we have exceptions such as the `ci` role and `breakglass` user as you have done.

For the desired effect you're going for just add the `administrator` role to the exceptions

File: data.tf [hunk 3]
```diff
@@ -56,34 +54,47 @@ data "aws_iam_policy_document" "deny_deletion_specific_resources" {
...
 
-    actions = [
-      "ec2:DeleteFlowLogs",
-      "logs:DeleteLogGroup",
-      "logs:DeleteLogStream"
-    ]
+    condition {
+      test     = "ArnNotLike"
+      variable = "aws:PrincipalARN"
+      values = [
```
Comment (djsd123) [inline data.tf:83]:
```suggestion
      values = [
      "arn:aws:iam::${each.value}:role/administrator",
      "arn:aws:iam::${each.value}:role/ci",
       "arn:aws:iam::${each.value}:user/breakglass"
```

File: locals.tf (hunk 1 above)
Comment (tea-short) [reply locals.tf]: Refactored to use the list of accounts from a resource attribute so the list doesn't have to be maintained anymore.

File: locals.tf (hunk 2 above)
Comment (tea-short) [reply locals.tf]: Nice, thanks, refactored

File: data.tf (hunk 3 above)
Comment (tea-short) [reply data.tf:78]: Refactored

**Review by djsd123 — APPROVED** [review]
💯

**Pull request merged**